GVRf/Framework/src/org/gearvrf/GVRResourceVolume.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import android.os.Environment;
+
+/**
+ * Models a file system which supports stream I/O.
```

Comment:
Let me point out something different from meshes and models to begin with (which is necessary):

1) Mesh and texture cannot include other files. -- so only a wrapper for a File would suffice -- this is what GVRAndroidResource is designed for. The newly added overload GVRAndroidResource(URL) is a compatible addition to original design and purpose. It can be used for texture / mesh as well.

2) Models require something that supports "openFile" (see IOSystem interface in assimp). FBX can include submodels in other FBX file. OBJ include MTL as a separate file. If we don't break original design of GVRAndroidResource, we would need an equivalent of IOSystem, for which I think the cheapest impl. will be something like the current GVRResourceVolume.